cmd/createFgaModel.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2024 NAME HERE <EMAIL ADDRESS>
+*/
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	authz "github.com/canonical/identity-platform-login-ui/internal/authorization"
+	logging "github.com/canonical/identity-platform-login-ui/internal/logging"
+	monitoring "github.com/canonical/identity-platform-login-ui/internal/monitoring"
+	fga "github.com/canonical/identity-platform-login-ui/internal/openfga"
+	tracing "github.com/canonical/identity-platform-login-ui/internal/tracing"
+	"github.com/spf13/cobra"
+)
+
+// createFgaModelCmd represents the createFgaModel command
+var createFgaModelCmd = &cobra.Command{
+	Use:   "create-fga-model",
+	Short: "Creates an openfga model",
+	Long:  `Creates an openfga model`,
+	Run: func(cmd *cobra.Command, args []string) {
+		apiUrl, _ := cmd.Flags().GetString("api-url")
+		apiToken, _ := cmd.Flags().GetString("api-token")
+		storeId, _ := cmd.Flags().GetString("store-id")
+		createModel(apiUrl, apiToken, storeId)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(createFgaModelCmd)
+
+	createFgaModelCmd.PersistentFlags().String("api-url", "", "The openfga API URL")
+	createFgaModelCmd.PersistentFlags().String("api-token", "", "The openfga API token")
+	createFgaModelCmd.PersistentFlags().String("store-id", "", "The openfga store to create the model in")
+	createFgaModelCmd.MarkPersistentFlagRequired("api-url")
+	createFgaModelCmd.MarkPersistentFlagRequired("api-token")
+	createFgaModelCmd.MarkPersistentFlagRequired("store-id")
+}
+
+func createModel(apiUrl, apiToken, storeId string) {
+	logger := logging.NewNoopLogger()
+	tracer := tracing.NewNoopTracer(nil)
+	monitor := monitoring.NewNoopMonitor("", logger)
+	scheme, host, err := parseURL(apiUrl)
+	if err != nil {
+		panic(err)
+	}
+	cfg := fga.NewConfig(scheme, host, storeId, apiToken, "", false, tracer, monitor, logger)
+	authzClient := fga.NewClient(cfg)
+	authorizer := authz.NewAuthorizer(authzClient, tracer, monitor, logger)
+	modelId, err := authorizer.CreateModel(context.Background())
```

Comment:
instead of enhancing the Authorizer, i would probably put it all inline here

https://github.com/canonical/identity-platform-login-ui/pull/215/files#diff-37ac464c45e06bed5132281126e435b869de3ca34bbdfb008db79edad30c8edcR60-R61

although if reasoning is to keep the logic and model insync happy to be proven wrong and keep it as is

Reply:
The reason is to use the model, which is at https://github.com/canonical/identity-platform-login-ui/blob/main/internal/authorization/auth_model.go. I could make the variable global and use that.